### PR TITLE
libvmi: random name for cloudinit volume

### LINKED ISSUE #12535: libvmi: constant volume name overwrite volume with the same name
**What happened**:
A volume with name `disk1` is always overwritten if cloudinit is added using libvmi or cloudinit is skipped if a volume with the name `disk1` was already added

**What you expected to happen**:
Be able to add a volume with name `disk1` independently form cloudinit

The current problem is that we use a constant name for the cloudinit volume (see [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/libvmi/cloudinit.go#L28))

### PR DESCRIPTION
**Pull request by alicefr**
If there is another volume using the name `disk1`, then it will conflict with the cloudinit constant name. Therefore, this change adds the random generation of the cloudinit name of the volume for libvmi.

### What this PR does
Before this PR:
Constant name `disk1` for the cloudinit volume for libvmi

After this PR:
Random suffix for the cloudinit volume

Fixes #12535

### Why we need it and why it was done in this way
The following tradeoffs were made:

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — COMMENTED** [review]
I'm fine with doing this.

But we do need to be sure this does not have side-effects: Some tests, may get broken when fixing it. i.e. some may have depended on the name to be fixed.
I guess that if all tests are passing, we should be good.

Can someone from sig-storage that is familiar with the e2e tests lgtm this?

File: pkg/libvmi/cloudinit/BUILD.bazel [hunk 1]
```diff
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
```
[the remark this answers is not in the document]
Comment (alicefr) [reply pkg/libvmi/cloudinit/BUILD.bazel]: Yes, sorry for the noise

**Comment by alicefr** [comment]
> I'm fine with doing this.
> 
> But we do need to be sure this does not have side-effects: Some tests, may get broken when fixing it. i.e. some may have depended on the name to be fixed. I guess that if all tests are passing, we should be good.
> 

Could be, but I doubt a little since the name of the volume is quite hidden. Anyway, let's see if the CI is happy

> Can someone from sig-storage that is familiar with the e2e tests lgtm this?

@mhenriks

**Comment by alicefr** [comment]
> > I'm fine with doing this.
> > But we do need to be sure this does not have side-effects: Some tests, may get broken when fixing it. i.e. some may have depended on the name to be fixed. I guess that if all tests are passing, we should be good.
> 
> Could be, but I doubt a little since the name of the volume is quite hidden. Anyway, let's see if the CI is happy
> 

Nevermind, it seems we have hardcoded the name of the disk for certain tests.

**Review by 0xFelix — COMMENTED** [review]
Alternative suggestion: Change the name of the disk to `cloudinitdisk`? That allows avoiding naming collisions and tests can still expect a deterministic name.

This name is also used in https://github.com/kubevirt/common-templates

**Comment by alicefr** [comment]
> Alternative suggestion: Change the name of the disk to `cloudinitdisk`? That allows avoiding naming collisions and tests can still expect a deterministic name.
> 
> This name is also used in https://github.com/kubevirt/common-templates

I can try, but I think if the tests are failing then they expect the name disk1

**Comment by xpivarc** [comment]
This reminds me https://github.com/kubevirt/kubevirt/pull/11005

**Comment by alicefr** [comment]
@xpivarc you can take a look to the linked issue, I also face a weird behavior where the disk wasn't added or cloudinit causes a panic for a nil pointer.

**Comment by alicefr** [comment]
@xpivarc would be fine for you this change? The `cloudinitdisk` name should be pretty safe

**Comment by xpivarc** [comment]
> @xpivarc would be fine for you this change? The `cloudinitdisk` name should be pretty safe

Sounds good to me.

**Review by 0xFelix — COMMENTED** [review]
/lgtm

**Comment by alicefr** [comment]
@xpivarc do you mind approving this?

**Pull request merged**